src/SessionManager.h

```diff
@@ -0,0 +1,242 @@
+	[[deprecated("Remove in v5.1. Use SessionMemoryUsageVals().")]]
+	unsigned int ConnectionMemoryUsageConnVals() { return SessionMemoryUsageVals(); }
+
+	unsigned int MemoryAllocation();
+
+	// TODO: should this move somewhere else?
+	analyzer::tcp::TCPStateStats tcp_stats;	// keeps statistics on TCP states
+
+private:
+
+	class StatBlocks {
+
+	public:
+
+		struct Block {
+			telemetry::IntGauge num;
+			telemetry::IntCounter total;
+			size_t max = 0;
+
+			Block(telemetry::IntGaugeFamily num_family,
+			      telemetry::IntCounterFamily total_family,
+			      std::string protocol) : num(num_family.GetOrAdd({{"protocol", protocol}})),
+			                              total(total_family.GetOrAdd({{"protocol", protocol}}))
+				{
+				}
+			};
+
+		using BlockMap = std::map<std::string, Block>;
+
+		BlockMap::iterator InitCounters(std::string protocol)
+			{
+			telemetry::IntGaugeFamily num_family = telemetry_mgr->GaugeFamily(
+				"zeek", "open-sessions", {"protocol"}, "Active Zeek Sessions");
+			telemetry::IntCounterFamily total_family = telemetry_mgr->CounterFamily(
+				"zeek", "sessions", {"protocol"},
+				"Total number of sessions", "1", true);
+
+			auto [it, inserted] = entries.insert(
+				{protocol, Block{num_family, total_family, protocol}});
+
+			if ( inserted )
+				return it;
+
+			return entries.end();
+			}
+
+		Block* GetCounters(std::string protocol)
+			{
+			auto it = entries.find(protocol);
+			if ( it == entries.end() )
+				it = InitCounters(protocol);
+
+			if ( it != entries.end() )
+				return &(it->second);
+
+			return nullptr;
+			}
+
+	private:
+
+		BlockMap entries;
+	};
+
+	using SessionMap = std::map<detail::SessionKey, Session*>;
+
+	Connection* NewConn(const detail::ConnIDKey& k, double t, const ConnID* id,
+	                    const u_char* data, int proto, uint32_t flow_label,
+	                    const Packet* pkt);
+
+	// Returns true if the port corresonds to an application
+	// for which there's a Bro analyzer (even if it might not
+	// be used by the present policy script), or it's more
+	// generally a likely server port, false otherwise.
+	//
+	// Note, port is in host order.
+	bool IsLikelyServerPort(uint32_t port, TransportProto transport_proto) const;
+
+	// Upon seeing the first packet of a connection, checks whether
+	// we want to analyze it (e.g., we may not want to look at partial
+	// connections), and, if yes, whether we should flip the roles of
+	// originator and responder (based on known ports or such).
+	// Use tcp_flags=0 for non-TCP.
+	bool WantConnection(uint16_t src_port, uint16_t dest_port,
+	                    TransportProto transport_proto,
+	                    uint8_t tcp_flags, bool& flip_roles);
+
+	// For a given protocol, checks whether the header's length as derived
+	// from lower-level headers or the length actually captured is less
+	// than that protocol's minimum header size.
+	bool CheckHeaderTrunc(int proto, uint32_t len, uint32_t caplen, const Packet *pkt);
```

Comment:
This doesn't fit either, but I'm not sure where it would go ...

Reply:
This also moves with the analyzer work.